Data Prepper 2.8.0 release blog post. (#2867)
opensearch-project/project-website: _posts/2024-05-16-Announcing-Data-Prepper-2.8.0.md

---
layout: post
title:  "Announcing Data Prepper 2.8.0"
authors:

> - qichen

Author: Fix alias: https://opensearch.org/community/members/george-chen.html

Irrespective of the current setting of the max message size, it is possible for the external source to send messages larger than the current max message size and Data Prepper fails to send that message to Kafka.
To prevent this, Data Prepper is modified to split the incoming messages to a size that is slightly smaller than the max message size before it is sent to Kafka.
Each message chunk is stand-alone message such a way that the consumer of the message can start processing it immediately instead of waiting for all chunks to be received.
This chunking of messages is currently supported only for JSON messages received by http source. Support for large OTEL metrics/traces/logs will be added in a future release.

## Other features

* The `s3` sink can write events to [Amazon S3](https://aws.amazon.com/s3/) objects with data from events in the key paths. This allows you to organize your key prefixes by dates or other metadata. This allows tools such as [Apache Hadoop](https://hadoop.apache.org/) and [Amazon Athena](https://aws.amazon.com/athena/) to more efficiently perform ad-hoc queries on your data.

> * Data Prepper now has an `ndjson` codec you can use to read data from S3 that looks like ndjson. By default, it is far more lenient and does not require newlines between JSON objects.

Author: I've normally see "newline" as a single word for these purposes. Wikipedia has this as well: https://en.wikipedia.org/wiki/Newline

> * Data Prepper now has an `ndjson` codec you can use to read data from S3 that looks like ndjson. By default, it is far more lenient and does not require newlines between JSON objects.

Reviewer: Ok, works for me 😄

* The new `write_json` processor can serialize event data into a JSON string that you can send downstream.
* The `s3` sink supports bucket ownership validation for more secure interactions with Amazon S3.

> * A new `typeof` operator allows you to write conditions which check the type of a field. You can use this anywhere you use other Data Prepper expressions such as in routes or when conditions.

Reviewer: ```suggestion
* A new `typeof` operator allows you to write conditions that check the type of a field. You can use this anywhere you use other Data Prepper expressions, such as in routes or conditions.
```
